### DISCUSSION TITLE
Is it possible to use API routes as normal pages?

### DISCUSSION THREAD

**Discussion by eddeee888** [vercel/next.js#30213] [Help] [answered]
Hello everyone,

I've got a use case where I want to use the  [API  routes](https://nextjs.org/docs/api-routes/introduction) syntax in the home page i.e. `/pages/index.ts`

Is this possible with Next.js ? Maybe through some custom proxy/config?

**Comment by gaston-flores** [comment, marked as answer]
You can use [rewrites](https://nextjs.org/docs/api-reference/next.config.js/rewrites) to do this:

```js
// next.config.js
module.exports = {
  async rewrites() {
    return [
      {
        source: "/",
        destination: "/api/your-api"
      }
    ];
  }
};
```

Note that for the above config to work, you will need to remove or rename your `/pages/index.ts` file. [Here's an example.](https://codesandbox.io/s/next-rewrite-api-index-kjbw7)

**Reply by eddeee888** [reply]
Awesome! This is perfect for my use case. Thank you very much! 🙏
